test/WebJobs.Script.Tests/TestScripts/CSharp/AssembliesFromSharedLocation/run.csx

```diff
@@ -0,0 +1,9 @@
+﻿#r "..\SharedAssemblies\PrimaryDependency.dll"
```

Comment:
We're OK violating the "don't check binaries in" because we're never going to alter the contents of these binaries? Also, the source for the binaries are just something you crufted up but doesn't exist in the repo? Another possible strategy might be to have the source also checked in and build as part of the tests, with the build output for them set to the shared location you want them in?